bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HttpHelper.java

```diff
@@ -60,6 +60,8 @@
     private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
     private static final JsonParser JSON_PARSER = new JsonParser();
     private static final Map<String, Bucket> BUCKET_MAP = new HashMap<>();
+    private static final Object AUTHORIZATION_HEADER_MONITOR = new Object();
+    private static final Object BUCKET_MONITOR = new Object();
```

Comment:
As these are static, the synchronization will be global across all bridges. Is that your intention or do you want to synchronize only the `oAuthClientService`?

Also, the bucket is valid for all bridges as the bucket map is static. IIRC, this was a topic when reviewing the initial contribution of this binding, but I can't remember the outcome of the discussion. Do you have that in mind?

Reply:
> As these are static, the synchronization will be global across all bridges. Is that your intention or do you want to synchronize only the oAuthClientService?

That is the intentions. `AUTHORIZATION_HEADER_MONITOR` needed to prevent that two threads renew the token at the "same" time. If you have more than one appliance this can happen. As it only happens every 24h (ttl of access token), I think it is ok synchronize it.

>Also, the bucket is valid for all bridges as the bucket map is static. IIRC, this was a topic when reviewing the initial contribution of this binding, but I can't remember the outcome of the discussion. Do you have that in mind?

Not sure. The bucket map contains a bucket entry for every bridge.

Reply:
Sounds reasonable!